<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>学生成绩与分析系统</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabData">
       <attribute name="title">
        <string>数据管理</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QGroupBox" name="groupBoxInput">
          <property name="title">
           <string>学生成绩录入</string>
          </property>
          <layout class="QFormLayout" name="formLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>学号:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="editStudentId"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>姓名:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="editStudentName"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>班级:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QComboBox" name="comboClass">
             <item>
              <property name="text">
               <string>2023级1班</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>2023级2班</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>2023级3班</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>课程:</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QComboBox" name="comboCourse">
             <item>
              <property name="text">
               <string>数学</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>语文</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>英语</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>物理</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>化学</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>成绩:</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QDoubleSpinBox" name="spinScore">
             <property name="maximum">
              <double>150.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.500000000000000</double>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>考试时间:</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QDateEdit" name="dateExam">
             <property name="calendarPopup">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QPushButton" name="btnAdd">
               <property name="text">
                <string>添加记录</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnUpdate">
               <property name="text">
                <string>更新记录</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnDelete">
               <property name="text">
                <string>删除记录</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="7" column="0" colspan="2">
            <widget class="QGroupBox" name="groupBoxBatch">
             <property name="title">
              <string>批量导入</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <item>
               <widget class="QPushButton" name="btnImportCSV">
                <property name="text">
                 <string>导入CSV文件</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBoxData">
          <property name="title">
           <string>成绩数据</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QPushButton" name="btnRefresh">
               <property name="text">
                <string>刷新</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="labelFilter">
               <property name="text">
                <string>筛选:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboFilterClass">
               <item>
                <property name="text">
                 <string>所有班级</string>
                </property>
               </item>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboFilterCourse">
               <item>
                <property name="text">
                 <string>所有课程</string>
                </property>
               </item>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboFilterDate">
               <item>
                <property name="text">
                 <string>所有日期</string>
                </property>
               </item>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="editSearch">
               <property name="placeholderText">
                <string>搜索学号或姓名...</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Orientation::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="btnExport">
               <property name="text">
                <string>导出报表</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QTableView" name="tableView">
             <property name="alternatingRowColors">
              <bool>true</bool>
             </property>
             <property name="selectionBehavior">
              <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="labelRecordCount">
             <property name="text">
              <string>总记录数: 0</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignmentFlag::AlignRight</set>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabStatistics">
       <attribute name="title">
        <string>统计分析</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QGroupBox" name="groupBoxStatsControl">
          <property name="title">
           <string>统计选项</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>选择班级:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboStatsClass">
             <item>
              <property name="text">
               <string>所有班级</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>选择课程:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboStatsCourse">
             <item>
              <property name="text">
               <string>所有课程</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="labelStatsDate">
             <property name="text">
              <string>考试日期:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboStatsDate">
             <item>
              <property name="text">
               <string>所有日期</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnCalculateStats">
             <property name="text">
              <string>计算统计</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnGenerateReport">
             <property name="text">
              <string>生成详细报告</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Orientation::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBoxStatsResult">
             <property name="title">
              <string>统计结果</string>
             </property>
             <layout class="QFormLayout" name="formLayout_2">
              <item row="0" column="0">
               <widget class="QLabel" name="labelAvg">
                <property name="text">
                 <string>平均分:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLabel" name="labelAvgValue">
                <property name="text">
                 <string>-</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="labelMax">
                <property name="text">
                 <string>最高分:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QLabel" name="labelMaxValue">
                <property name="text">
                 <string>-</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="labelMin">
                <property name="text">
                 <string>最低分:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QLabel" name="labelMinValue">
                <property name="text">
                 <string>-</string>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QLabel" name="labelStdDev">
                <property name="text">
                 <string>标准差:</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QLabel" name="labelStdDevValue">
                <property name="text">
                 <string>-</string>
                </property>
               </widget>
              </item>
              <item row="4" column="0">
               <widget class="QLabel" name="labelPassRate">
                <property name="text">
                 <string>及格率:</string>
                </property>
               </widget>
              </item>
              <item row="4" column="1">
               <widget class="QLabel" name="labelPassRateValue">
                <property name="text">
                 <string>-</string>
                </property>
               </widget>
              </item>
              <item row="5" column="0">
               <widget class="QLabel" name="labelCount">
                <property name="text">
                 <string>学生数:</string>
                </property>
               </widget>
              </item>
              <item row="5" column="1">
               <widget class="QLabel" name="labelCountValue">
                <property name="text">
                 <string>-</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QTabWidget" name="tabWidgetCharts">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="tabChart1">
           <attribute name="title">
            <string>分数分布</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <widget class="QChartView" name="chartViewHistogram" native="true"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tabChart2">
           <attribute name="title">
            <string>趋势分析</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <widget class="QChartView" name="chartViewTrend" native="true"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tabChart3">
           <attribute name="title">
            <string>对比分析</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_8">
            <item>
             <widget class="QChartView" name="chartViewComparison" native="true"/>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>18</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="actionImport"/>
    <addaction name="actionExport"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>数据</string>
    </property>
    <addaction name="actionAddRecord"/>
    <addaction name="actionEditRecord"/>
    <addaction name="actionDeleteRecord"/>
    <addaction name="separator"/>
    <addaction name="actionRefresh"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>分析</string>
    </property>
    <addaction name="actionStatistics"/>
    <addaction name="actionCharts"/>
    <addaction name="actionReports"/>
   </widget>
   <widget class="QMenu" name="menu_4">
    <property name="title">
     <string>帮助</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
   <addaction name="menu_3"/>
   <addaction name="menu_4"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionImport">
   <property name="text">
    <string>导入数据</string>
   </property>
  </action>
  <action name="actionExport">
   <property name="text">
    <string>导出报表</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>退出</string>
   </property>
  </action>
  <action name="actionAddRecord">
   <property name="text">
    <string>添加记录</string>
   </property>
  </action>
  <action name="actionEditRecord">
   <property name="text">
    <string>编辑记录</string>
   </property>
  </action>
  <action name="actionDeleteRecord">
   <property name="text">
    <string>删除记录</string>
   </property>
  </action>
  <action name="actionRefresh">
   <property name="text">
    <string>刷新数据</string>
   </property>
  </action>
  <action name="actionStatistics">
   <property name="text">
    <string>统计分析</string>
   </property>
  </action>
  <action name="actionCharts">
   <property name="text">
    <string>图表分析</string>
   </property>
  </action>
  <action name="actionReports">
   <property name="text">
    <string>生成报告</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>关于</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QChartView</class>
   <extends>QWidget</extends>
   <header>qchartview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
